src/Analyzers/Core/Analyzers/UseCollectionInitializer/AbstractUseCollectionInitializerDiagnosticAnalyzer.cs

```diff
@@ -132,29 +144,96 @@ private void AnalyzeNode(SyntaxNodeAnalysisContext context, INamedTypeSymbol ien
                 objectCreationExpression.GetFirstToken().GetLocation(),
                 option.Notification.Severity,
                 additionalLocations: locations,
-                properties: null));
+                properties: shouldUseCollectionExpression ? UseCollectionInitializerHelpers.UseCollectionExpressionProperties : null));
+
+            FadeOutCode(context, matches, locations);
+
+            return;
+
+            (ImmutableArray<Match<TStatementSyntax>> matches, bool shouldUseCollectionExpression) GetMatches()
```

Comment:
new logic that first tries to see if we can convert to a collection expr.  otherwise, falls back to original approach of trying a collection-initializer.  

allows us to share the majorty of code and concepts.